js/src/logger.ts

```diff
@@ -490,13 +490,14 @@ export class BraintrustState {
     if (this.apiUrl && !loginParams.forceLogin) {
       return;
     }
-    const newState = await loginToState({
+    const { state: newState, conn } = await loginToState({
       ...this.loginParams,
       ...Object.fromEntries(
         Object.entries(loginParams).filter(([k, v]) => !isEmpty(v)),
       ),
     });
     this.copyLoginInfo(newState);
+    this.loginReplaceApiConn(conn);
```

Comment:
I wonder if instead we should call `this.loginReplaceApiConn` [here](https://github.com/braintrustdata/braintrust-sdk/blob/b432e3b9926b4b379abe068cf156dfc4060810b3/js/src/logger.ts#L410) and leave this as-is.